backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/question/domain/MultipleChoiceQuestion.java

```diff
@@ -0,0 +1,103 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.question.domain;
+
+import pt.ulisboa.tecnico.socialsoftware.tutor.answer.domain.QuestionAnswer;
+import pt.ulisboa.tecnico.socialsoftware.tutor.config.DateHandler;
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.Course;
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.TutorException;
+import pt.ulisboa.tecnico.socialsoftware.tutor.impexp.domain.Visitor;
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.OptionDto;
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.dto.QuestionDto;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.ONE_CORRECT_OPTION_NEEDED;
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.OPTION_NOT_FOUND;
+
+@Entity
+@DiscriminatorValue(Question.QuestionTypes.MultipleChoice)
+public class MultipleChoiceQuestion extends Question {
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "question",fetch = FetchType.EAGER, orphanRemoval=true)
+    private final List<Option> options = new ArrayList<>();
+
+
+    public List<Option> getOptions() {
+        return options;
+    }
+
+    public MultipleChoiceQuestion(){
+
+    }
+
+    public MultipleChoiceQuestion(Course course, QuestionDto questionDto) {
+        super(course, questionDto);
+        setOptions(questionDto.getOptions());
+    }
+
+    public void setOptions(List<OptionDto> options) {
+        if (options.stream().filter(OptionDto::getCorrect).count() != 1) {
+            throw new TutorException(ONE_CORRECT_OPTION_NEEDED);
+        }
+
+        int index = 0;
+        for (OptionDto optionDto : options) {
+            if (optionDto.getId() == null) {
+                optionDto.setSequence(index++);
+                new Option(optionDto).setQuestion(this);
+            } else {
+                Option option = getOptions()
+                        .stream()
+                        .filter(op -> op.getId().equals(optionDto.getId()))
+                        .findFirst()
+                        .orElseThrow(() -> new TutorException(OPTION_NOT_FOUND, optionDto.getId()));
+
+                option.setContent(optionDto.getContent());
+                option.setCorrect(optionDto.getCorrect());
+            }
+        }
+    }
+
+    public void addOption(Option option) {
+        options.add(option);
+    }
+
+    public Integer getCorrectOptionId() {
+        return this.getOptions().stream()
+                .filter(Option::getCorrect)
+                .findAny()
+                .map(Option::getId)
+                .orElse(null);
+    }
+
+    public void update(QuestionDto questionDto) {
+        super.update(questionDto);
+        setOptions(questionDto.getOptions());
+    }
+
+    @Override
+    public void visitDependencies(Visitor visitor) {
```

Comment:
should follow how it is implemented in the other domain entities, 
accept(Visitor visitor) { visitor.visitMultipleChoiceQuestion(this);}